core/shared/platform/common/posix/posix_socket.c

```diff
@@ -351,6 +353,473 @@ os_socket_convert_sockaddr(struct sockaddr *addr, uint8_t *buf, size_t buflen,
     return BHT_OK;
 }
 
+int
```

Comment:
should that be a static function or are you going to export it from the unit and use somewhere else as well?